File: backend/src/main/java/etna/tavernn/club/model/Club.java
package etna.tavernn.club.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "clubs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Club {

    @Id
    @Column(length = 36)
    private String id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "creation_date", nullable = false)
    private LocalDate creationDate;

    @Column(name = "logo")
    private String logo;

    @Enumerated(EnumType.STRING)
    @Column(name = "club_type", nullable = false)
    private ClubType clubType = ClubType.open;

    @Column(name = "nr_members", nullable = false)
    private Integer nrMembers = 0;

    @Column(name = "max_members", nullable = false)
    private Integer maxMembers;

    @OneToMany(mappedBy = "club", cascade = CascadeType.ALL)
    @Builder.Default
    private List<ClubMember> members = new ArrayList<>();
}

File: frontend/src/types/club.ts
import {ClubMember} from "@/types/clubMember.ts";
import {Team} from "@/types/team.ts";

export interface Club {
    id: string;
    name: string;
    description: string | null;
    creationDate: string;
    logo: string | undefined;
    clubType: 'open' | 'closed' | 'invite_only';
    nrMembers: number;
    nrTeams: number;
    maxMembers: number;
    owner: ClubMember;
    admins: ClubMember[];
    members: ClubMember[];
    teams: Team[];
}

export interface ClubRequest {
    name: string;
    description: string | null;
    clubType: string;
    maxMembers: number;
}